File: 226.invert-binary-tree.py
#
# @lc app=leetcode id=226 lang=python
#
# [226] Invert Binary Tree
#
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if not root: return None
        
        import collections
        queue = collections.deque([root])

        while queue:
            cur_node = queue.popleft()

            cur_node.left, cur_node.right = cur_node.right, cur_node.left
            if cur_node.left: queue.append(cur_node.left)        
            if cur_node.right: queue.append(cur_node.right)        
        
        return root

File: 226.invert-binary-tree.java
/*
 * @lc app=leetcode id=226 lang=java
 *
 * [226] Invert Binary Tree
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode node) {
        if (node == null) {
            return null;
        }
        
        TreeNode tmp = invertTree(node.left);
        node.left = invertTree(node.right);
        node.right = tmp;
        
        return node;
    }
}
// @lc code=end

